packages/frontend/src/components/accounts/AccountFormAccountId.js

```diff
@@ -50,229 +51,208 @@ const InputWrapper = styled.div`
         const extraSpace = isSafari ? 21.5 : 22;
-        this.suffix.current.style.left = width + extraSpace + 'px';
-        this.suffix.current.style.visibility = 'visible';
+        suffix.current.style.left = `${width + extraSpace}px`;
+        suffix.current.style.visibility = 'visible';
         if (userValue.length === 0) {
-            this.suffix.current.style.visibility = 'hidden';
+            suffix.current.style.visibility = 'hidden';
         }
-    }
+    };
 
-    getTextWidth = (text, font) => {
-        if (!this.canvas) {
-            this.canvas = document.createElement('canvas');
+    const getTextWidth = (text, font) => {
+        if (!canvas.current) {
+            canvas.current = document.createElement('canvas');
         }
-        let context = this.canvas.getContext('2d');
+        const context = canvas.current.getContext('2d');
         context.font = font;
-        let metrics = context.measureText(text);
+        const metrics = context.measureText(text);
         return metrics.width;
-    }
+    };
 
-    handleChangeAccountId = ({ userValue, el }) => {
-        const { pattern, handleChange, type } = this.props;
+    const handleChangeAccountId = ({ userValue, el }) => {
+        const currentAccountId = userValue.toLowerCase();
 
-        const accountId = userValue.toLowerCase();
-
-        if (accountId === this.state.accountId) {
+        if (currentAccountId === accountId) {
             return;
         }
 
-        if (accountId.match(pattern)) {
-            if (this.state.wrongChar) {
+        if (currentAccountId.match(props.pattern)) {
+            if (wrongChar) {
                 el.style.animation = 'none';
                 void el.offsetHeight;
                 el.style.animation = null;
             } else {
-                this.setState(() => ({
-                    wrongChar: true
-                }));
+                setWrongChar(true);
             }
             return false;
         } else {
-            this.setState(() => ({
-                wrongChar: false
-            }));
+            setWrongChar(false);
         }
 
-        this.setState(() => ({
-            accountId: accountId
-        }));
-
-        handleChange(accountId);
-
-        this.props.localAlert && this.props.clearLocalAlert();
-
-        this.state.invalidAccountIdLength && this.handleAccountIdLengthState(accountId);
+        setAccountId(currentAccountId);
+        props.handleChange(currentAccountId);
+        props.localAlert && props.clearLocalAlert();
+        invalidAccountIdLength && handleAccountIdLengthState(currentAccountId);
+    };
 
-        this.checkAccountAvailabilityTimer && clearTimeout(this.checkAccountAvailabilityTimer);
-        this.checkAccountAvailabilityTimer = setTimeout(() => {
-            this.handleCheckAvailability(accountId, type);
-        }, ACCOUNT_CHECK_TIMEOUT);
-    }
-
-    checkAccountIdLength = (accountId) => {
+    const checkAccountIdLength = (accountId) => {
```

Comment:
Don't like the naming of this function. For the boolean response, what question is this function asking? Its ambiguous here.

Reply:
I just left the function names as they were previously, but have changed the name to be more descriptive.